### DISCUSSION TITLE
Draco breaks the geometry of long objects

### DISCUSSION THREAD

**Discussion by NeluQi** [donmccurdy/glTF-Transform#269] [Q&A] [answered]
After Draco squeezed, the geometry of the long objects broke.

![image](https://user-images.githubusercontent.com/16963395/119712856-b906b480-be69-11eb-8e27-9df50d4e6231.png)

![image](https://user-images.githubusercontent.com/16963395/119713395-53ff8e80-be6a-11eb-9524-b17a4f14d66f.png)

I will attach the original and compressed models
https://drive.google.com/file/d/1lGL_0los-PoSkNo7xAWK08aJOWElC2J0/view?usp=sharing

**Comment by donmccurdy** [comment, marked as answer]
Hi @NeluQi, thanks for reaching out!

Draco is a "lossy" compression algorithm, and tries approximate the input geometry without losing too much detail. Part of that process involves quantization (see [this Draco explanation](https://gamedev.stackexchange.com/a/175215/122501)). The default setting for the vertex positions is 14 bits, which means vertices are snapped to a 16K x 16K x 16K grid (2^14 in each dimension). A very large object with very tiny details is going to be more difficult to quantize, because the details are small and the 16K grid is stretched over a large volume.

I think there are two ways to reduce the problem you're seeing here:

- **A:** Divide the mesh into smaller parts. Currently this is one long mesh. Dividing it into smaller parts allows the quantization grid (which is separate for each part) to provide more detail. The tradeoff here is that having too many parts (e.g. >100) will start to reduce rendering framerate.
- **B:** Increase quantization bits. The default is 14 bits, but using 16-24 bits for vertex positions will allow more detail. The tradeoff here is that higher values reduce compression ratio somewhat. For example...

```
gltf-transform draco input.glb output.glb --quantize-position 24
```

... improves the results, but the final size will be 46 MB → 3.23 MB instead of 46 MB → 1.6 MB.
